### PR TITLE
Automate lint and test with GitHub Actions

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Commits pushed to the `main` branch and commits belonging to pull requests will be checked for linter errors.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    branches: [main]
+  pull_request:
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: golangci/golangci-lint-action@v6
+        with:
+          only-new-issues: true
```
Comment (0x2b3bfa0) [inline .github/workflows/test.yml]: Consider removing this once existing issues are fixed.

```suggestion
```

Comment (0x2b3bfa0) [reply .github/workflows/test.yml:12]:
```suggestion
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+on:
+  push:
+    branches: [main]
+  pull_request:
```
Comment (martinmunillas) [inline .github/workflows/test.yml:4]: I would add it in PRs as well

Comment (0x2b3bfa0) [reply .github/workflows/test.yml:4]: Note that https://github.com/martinmunillas/otter/pull/3/commits/5d2438bd0c05cdd618f69ebbc1cc76156a979cf9 is superfluous: an empty `pull_request` YAML mapping already means that it'll run on pull requests targeting _any_ branch.

```yaml
on:
  pull_request:
  # implicit (catch-all)
```

```yaml
on:
  pull_request:
    # explicit (catch-all, same as above)
    branches: ['**']
```

```yaml
on:
  pull_request:
    # explicit (run only for pull requests targeting the main branch)
    branches: [main]
```

> _[You can use the `branches` or `branches-ignore` filter to configure your workflow to **only** run on pull requests that target specific branches.](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request)_ (emphasis mine)

**Comment by 0x2b3bfa0** [comment]
> [!TIP]
> Consider using [squash merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits), especially for pull requests like mine, which don't follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines and whose commit-level history might not be particularly meaningful.

File: .github/workflows/test.yml (hunk 1 above)
Comment (0x2b3bfa0) [reply .github/workflows/test.yml]: * https://github.com/martinmunillas/otter/pull/4

**Pull request merged**